main/engines/categories.py

```diff
@@ -0,0 +1,50 @@
+from collections.abc import Sequence
+
+from sqlalchemy import delete, select
+from sqlalchemy.exc import SQLAlchemyError
+
+from main import db
+from main.commons.exceptions import BadRequest
+from main.models.category import CategoryModel
+
+
+async def get_categories() -> Sequence[CategoryModel]:
+    statement = select(CategoryModel)
+    result = await db.session.execute(statement)
+    return result.scalars().all()
+
+
+async def add_category(
+    name: str,
+    creator_id: int,
+) -> CategoryModel:
+    category = CategoryModel(name=name, creator_id=creator_id)
+
+    try:
+        db.session.add(category)
+        await db.session.commit()
+    except SQLAlchemyError:
+        raise BadRequest()
```

Comment:
BaseError is specific to return to the client and should only be raised in controllers. If you need to raise an exception from engines, create a custom exception class. Btw, there are a lot of places where you catch SQLAlchemyError and raise BadRequest. Should we do it in one place?

Reply:
Updated to handle `SQLAlchemyError` in `main/commons/error_handlers.py`.
This error should only be raised by edge-case scenarios: race conditions - 2 workers adding category with the same name to DB simultaneously for example. Therefore I'd go the a simple solution here.

Reply:
Removed, use 500 Internal server error for these cases.